run-test.py

```diff
@@ -838,13 +880,75 @@ def test_netsplit_single_virtual(pod, pod_ips):
     stop_test()
 
 
+# check_abandoned_requests calculates abandoned requests leak.
+#
+# nattempts - number of attempts for checking abandoned requests metric from nodes.
+# step - time in seconds between two attempts.
+# verbose - flag for additional logging.
+def check_abandoned_requests(nattempts=5, step=15, verbose=False):
+    start_test("check_abandoned_requests")
+    info("==== start/stop check_abandoned_requests test started ====")
+
+    # Dict with count of abandoned metric. (key - <node_and_metric_mane>, value - <count>).
+    # Example: <10.1.0.179:insolar_requests_abandoned{role="heavy_material"} 20>,
+    #          <10.1.0.180:insolar_requests_abandoned{role="light_material"} 35>,
+    #          ...
+    abandoned_data = {}
+    # Difference of abandoned requests count between two steps.
+    abandoned_delta = 0
+    errors = ""
+
+    for attempt in range(1, nattempts+1):
+        time.sleep(step)
+
+        abandoned_delta = 0
+        abandoned_raw_data = get_abandones_count_from_nodes()
+
+        if len(abandoned_raw_data) == 0:
+            continue
+
+        for line in abandoned_raw_data.split("\n"):
+            kv = line.split()
+            node = kv[0]        # key for abandoned_data dict.
+            count = int(kv[1])  # value for abandoned_data dict.
+            if node in abandoned_data and count > abandoned_data[node]:
+                abandoned_delta += count - abandoned_data[node]
+                errors += "Attempt: " + str(attempt) + "Abandoned increase in " + node + ". Old:" + str(abandoned_data[node]) + ", New:" + str(count) + "\n"
```

Comment:
I think that we should add node ID for possible investigations, easily we can get it from pod_ips